src/openqaoa-core/algorithms/qaoa/qaoa_result.py

```diff
@@ -161,43 +165,67 @@ def asdict(
         # if the backend is a statevector backend, the measurement outcomes will be the statevector,
         # meaning that it is a list of complex numbers, which is not serializable.
         # If that is the case, and complex_to_string is true the complex numbers are converted to strings.
-        if complex_to_string and issubclass(
-            self.__type_backend, QAOABaseBackendStatevector
-        ):
+        if complex_to_string:
             return_dict["intermediate"] = {}
             for key, value in self.intermediate.items():
-                if (
-                    intermediate_mesurements == False and "measurement" in key
-                ):  # if intermediate_mesurements is false, the intermediate measurements are not included in the dump
-                    return_dict["intermediate"][key] = []
-                elif "measurement" in key and (
-                    isinstance(value, list) or isinstance(value, np.ndarray)
+                # Measurements and Cost may require casting
+                if "measurement" in key:
```

Comment:
I am not very happy with this solution: we may want to improve on it and have a more systematic way to ensure that `numpy` variables are converted to Python primitives

Reply:
raised an issue to fix this in a future PR, https://github.com/entropicalabs/openqaoa/issues/228